src/military/military.cpp

```diff
@@ -4659,7 +4746,7 @@ void end_battle(sys::state& state, dcon::naval_battle_id b, battle_result result
 				auto em_regs = em.get_army().get_army_membership();
 				while(em_regs.begin() != em_regs.end() && transport_cap < 0) {
 					auto reg_id = (*em_regs.begin()).get_regiment();
-					disband_regiment_w_pop_death(state, reg_id);
+					state.world.delete_regiment(reg_id);
```

Comment:
Is there a reason that you have reverted "killing" regiments that are lost at sea?

Reply:
Reintroduced